### PR TITLE
Turn state

### PR DESCRIPTION
**Pull request by AWeidenkopf**
Hey, I moved the content in GAME.TSX to their on components

### CODE REVIEW

File: src/components/ToggleActorBtns.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import styles from "../scenes/Game.module.css";
+
+interface ToggleActorBtnsProps {
+  onToggleActorView: () => void;
+  psychicBtn: boolean;
+  playerBtn: boolean;
+}
+
+export const ToggleActorBtns = ({
```
Comment (scottysseus) [inline src/components/ToggleActorBtns.tsx:10]: Can you rename this component and file to `ActorToggle`

File: src/components/RebuttalBtns.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+
+interface RebuttalBtnsProps {
+  setRebuttal: (value: string) => void;
+}
+
+export const RebuttalBtns = ({ setRebuttal }: RebuttalBtnsProps) => {
```
Comment (scottysseus) [inline src/components/RebuttalBtns.tsx:7]: Can you rename this component and file to `RebuttalToggle`?

File: src/components/NewGameBtn.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+interface NewGameBtnProps {
+  onNewGameClick: () => void;
+}
+
+export const NewGameBtn = ({ onNewGameClick }: NewGameBtnProps) => {
```
Comment (scottysseus) [inline src/components/NewGameBtn.tsx:7]: can you rename this `NewGameButton`?

File: src/components/Cards.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import styles from "../scenes/Game.module.css";
+import { BsArrowLeftSquare, BsArrowRightSquare } from "react-icons/bs";
+import { SharedState } from "src/store/SharedState";
+
+interface CardsProps {
+  sharedState: SharedState;
+}
+
+export const Cards = ({ sharedState }: CardsProps) => {
```
Comment (scottysseus) [inline src/components/Cards.tsx:10]: can you rename this to `Spectrum`?

Comment (scottysseus) [inline src/components/Cards.tsx:10]: This component does not need the entire `SharedState`....can't it just accept a `Spectrum` as a prop?

File: src/components/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import styles from "../scenes/Game.module.css";
+import { SharedState } from "src/store/SharedState";
+
+interface HeaderProps {
+  sharedState: SharedState;
+}
+
+export const Header = ({ sharedState }: HeaderProps) => {
```
Comment (scottysseus) [inline src/components/Header.tsx:9]: This component just needs the score map, not the entire `SharedState`.

File: src/components/Hint.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  onUpdateHint: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onSubmitHint: () => void;
+}
+
+export const Hint = ({
+  player,
+  sharedState,
+  onSubmitHint,
+  onUpdateHint,
+}: HintProps) => {
```
Comment (scottysseus) [inline src/components/Hint.tsx:17]: you only need the hint, not the entire `SharedState`

File: src/components/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <>
+      <div className={styles.pageHeader}>
+        <h1>CHADBURN</h1>
+      </div>
+
+      <div className={styles.turnContainer}>
+        <h3>
+          Blue: {sharedState.game.score.get("blue")} Red:{" "}
```
Comment (scottysseus) [inline src/components/Header.tsx:18]: try getting rid of the `{" "}`

File: src/components/NewGameBtn.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <>
+      {" "}
+      <div>
```
Comment (scottysseus) [inline src/components/NewGameBtn.tsx:11]: is this `{" "}` necessary? I would also get rid of the `div` for now

File: src/scenes/Game.tsx [hunk 9]
```diff
@@ -35,6 +39,14 @@ export const Game = ({ sharedState, publish }: GameProps) => {
     }
   }, [sharedState]);
 
+  useEffect(() => {
+    console.log(sharedState.game.score);
```
Comment (scottysseus) [inline src/scenes/Game.tsx:43]: gasp!

File: src/game/game.ts [hunk 10]
```diff
@@ -139,3 +140,24 @@ export function getTeamOutOfTurn(state: GameState): string {
 export function updateTurn(state: GameState, turn: TurnState): GameState {
   return { ...state, turn };
 }
+
+export function checkForEndOfGame(state: SharedState): boolean {
```
Comment (scottysseus) [inline src/game/game.ts:144]: can you rename this `isGameOver` ?

File: src/scenes/RebuttalView.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  getTeamOutOfTurn: (state: GameState) => string;
+  sharedState: SharedState;
+  onSubmitRebuttal: () => void;
+  setRebuttal: (value: string) => void;
+}
+
+export const RebuttalView = ({
+  guessSubmitted,
+  getTeamOutOfTurn,
+  sharedState,
```
Comment (scottysseus) [inline src/scenes/RebuttalView.tsx:18]: this can just be `Game` state

**Review by scottysseus — CHANGES_REQUESTED** [review]

File: src/components/ToggleActorBtns.tsx (hunk 1 above)
Comment (AWeidenkopf) [reply src/components/ToggleActorBtns.tsx:10]: Done!

File: src/components/RebuttalBtns.tsx (hunk 2 above)
Comment (AWeidenkopf) [reply src/components/RebuttalBtns.tsx:7]: Done!

File: src/components/Cards.tsx (hunk 4 above)
Comment (AWeidenkopf) [reply src/components/Cards.tsx:10]: ok

File: src/scenes/Game.tsx (hunk 9 above)
Comment (AWeidenkopf) [reply src/scenes/Game.tsx:43]: hehehe

**Review by scottysseus — APPROVED** [review]

**Pull request merged**